server/src/main/java/keywhiz/service/permissions/AnyPermissionCheck.java

```diff
@@ -0,0 +1,34 @@
+package keywhiz.service.permissions;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AnyPermissionCheck implements PermissionCheck {
+
+  private static final Logger logger = LoggerFactory.getLogger(AnyPermissionCheck.class);
+
+  private List<PermissionCheck> subordinateChecks;
+
+  public AnyPermissionCheck(List<PermissionCheck> subordinateChecks) {
+    this.subordinateChecks = ImmutableList.copyOf(subordinateChecks);
+  }
+
+  public boolean isAllowed(Object source, String action, Object target) {
+    boolean hasPermission = false;
+
+    for (PermissionCheck subordinateCheck : subordinateChecks) {
```

Comment:
The functionality here looks great, so this is just food for thought/thinking about alternative ways to accomplish the same task: if all you're looking for is that one of the subordinate checks returns true (rather than a specific one) you could replace this with a steam like:

```
boolean hasPermission  = subordinateChecks.stream().anyMatch(subordinateCheck
                     -> subordinateCheck.isAllowed(source, action, target));
```

(look like the tradeoff here is that streams may not short circuit: https://stackoverflow.com/questions/32192846/java-8-stream-short-circuit#:~:text=The%20operations%20allMatch%20%2C%20noneMatch%20and,using%20these%20may%20not%20terminate.)

I personally like streams because I find them to be an intuitive way to think about logic - don't feel obligated to change this at all, especially since the method is pretty simple - everyone has their own coding style!

Reply:
I hadn't thought about using streams but will keep that in mind for next time!